### PR TITLE
[ImageFont] Fix escape sequence parsing, add note to the docs.

### LINKED ISSUE #95293: Kerning Pairs property for Bitmap Image Font Data uses Unicode internal values as characters
### Tested versions

Reproducible in - Godot v4.3.rc3 - Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3050 (NVIDIA; 31.0.15.5176) - AMD Ryzen 7 2700 Eight-Core Processor (16 Threads)

### System information

Godot v4.3.rc3 - Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3050 (NVIDIA; 31.0.15.5176) - AMD Ryzen 7 2700 Eight-Core Processor (16 Threads)

### Issue description

When setting up kerning pairs in the bitmap image font editor, Unicode values (added to support things like spaces, e.g. \u0020) will have their internal characters erroneously encoded as pairs.

The tokenizer needs something like `i+=5` in the loops to skip forward past the Unicode characters once decoded.

### Steps to reproduce

See Kerning pairs import property in attached file.
\0020 will have u, 0, and 2 get erroneously encoded as pairs.

### Minimal reproduction project (MRP)

[kerning_unicode_issue.zip](https://github.com/user-attachments/files/16552030/kerning_unicode_issue.zip)

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes https://github.com/godotengine/godot/issues/95293

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

File: doc/classes/ResourceImporterImageFont.xml [hunk 1]
```diff
@@ -40,7 +40,7 @@
 		</member>
 		<member name="kerning_pairs" type="PackedStringArray" setter="" getter="" default="PackedStringArray()">
 			Kerning pairs for the font. Kerning pair adjust the spacing between two characters.
-			Each string consist of three space separated values: "from" string, "to" string and integer offset. Each combination form the two string for a kerning pair, e.g, [code]ab cd -3[/code] will create kerning pairs [code]ac[/code], [code]ad[/code], [code]bc[/code], and [code]bd[/code] with offset [code]-3[/code].
+			Each string consist of three space separated values: "from" string, "to" string and integer offset. Each combination form the two string for a kerning pair, e.g, [code]ab cd -3[/code] will create kerning pairs [code]ac[/code], [code]ad[/code], [code]bc[/code], and [code]bd[/code] with offset [code]-3[/code]. [code]\uXXXX[/code] escape sequence can be used to add Unicode characters.
```
[the remark this answers is not in the document]
Comment (akien-mga) [reply doc/classes/ResourceImporterImageFont.xml:43]: My bad, didn't see the GitHub notification for this review before merging.

[the remark this answers is not in the document]
Comment (AThousandShips) [reply doc/classes/ResourceImporterImageFont.xml:43]: Doesn't seem this was merged yet!

[the remark this answers is not in the document]
Comment (akien-mga) [reply doc/classes/ResourceImporterImageFont.xml]: Oh right, so I _did_ notice :D

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 4.3.1.

**Pull request merged**